src/fswAlgorithms/attGuidance/prescribedRot2DOF/prescribedRot2DOF.c

```diff
@@ -87,4 +89,156 @@ are updated in this routine as a function of time and written to the prescribedM
 */
 void Update_prescribedRot2DOF(PrescribedRot2DOFConfig *configData, uint64_t callTime, int64_t moduleID)
 {
+    // Create buffer messages
+    HingedRigidBodyMsgPayload spinningBodyRef1In;
+    HingedRigidBodyMsgPayload spinningBodyRef2In;
+    HingedRigidBodyMsgPayload spinningBodyOut;
+    PrescribedMotionMsgPayload prescribedMotionOut;
+
+    // Zero the output messages
+    spinningBodyOut = HingedRigidBodyMsg_C_zeroMsgPayload();
+    prescribedMotionOut = PrescribedMotionMsg_C_zeroMsgPayload();
+
+    // Read the input messages
+    spinningBodyRef1In = HingedRigidBodyMsg_C_zeroMsgPayload();
+    if (HingedRigidBodyMsg_C_isWritten(&configData->spinningBodyRef1InMsg))
+    {
+        spinningBodyRef1In = HingedRigidBodyMsg_C_read(&configData->spinningBodyRef1InMsg);
+    }
+
+    spinningBodyRef2In = HingedRigidBodyMsg_C_zeroMsgPayload();
+    if (HingedRigidBodyMsg_C_isWritten(&configData->spinningBodyRef2InMsg))
+    {
+        spinningBodyRef2In = HingedRigidBodyMsg_C_read(&configData->spinningBodyRef2InMsg);
+    }
+
+    /* This loop is entered when the spinning body attitude converges to the reference attitude. The PRV angle and axis
+     reference parameters are updated along with the profiled trajectory parameters. */
+    if ((HingedRigidBodyMsg_C_timeWritten(&configData->spinningBodyRef1InMsg) <= callTime || HingedRigidBodyMsg_C_timeWritten(&configData->spinningBodyRef2InMsg) <= callTime ) && configData->convergence)
```

Comment:
:small_red_triangle_down: Drop the second clause on to a new line so it doesn't overflow 120 characters.

Reply:
Done.